From stern@rowland.harvard.edu  Mon Mar 15 16:34:38 2010
From: Alan Stern <stern@rowland.harvard.edu>
Date: Mon, 8 Mar 2010 16:46:19 -0500 (EST)
Subject: Driver core: don't initialize wakeup flags
To: Greg KH <greg@kroah.com>
Message-ID: <Pine.LNX.4.44L0.1003081645100.1297-100000@iolanthe.rowland.org>


This patch (as1351) removes an unnecessary and unwanted assignment
from device_initialize().  The wakeup flags are set to 0 along with
everything else when the device structure is allocated, so we don't
need to do it again.  Furthermore, the subsystem might already have
set these flags to their correct values; we don't want to override it.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/core.c |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/base/core.c
+++ b/drivers/base/core.c
@@ -562,7 +562,6 @@ void device_initialize(struct device *de
 	init_MUTEX(&dev->sem);
 	spin_lock_init(&dev->devres_lock);
 	INIT_LIST_HEAD(&dev->devres_head);
-	device_init_wakeup(dev, 0);
 	device_pm_init(dev);
 	set_dev_node(dev, -1);
 }
